hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieClient.java

```diff
@@ -48,6 +50,7 @@
   protected final transient Configuration hadoopConf;
   protected final HoodieWriteConfig config;
   protected final String basePath;
+  protected AtomicReference<Option<HoodieInstant>> latestWriteInstantCompletedBeforeWriter = new AtomicReference<>();
```

Comment:
AtomicReference to handle multiple threads using the same client object? +1

Reply:
Since this is specific to writing, can we put this in the AbstractHoodieWriteClient subclass?